client/lib/posts/utils.js

```diff
@@ -217,6 +209,34 @@ var utils = {
 	},
 
 	/**
+	 * Returns the ID of the featured image assigned to the specified post, or
+	 * `undefined` otherwise. A utility function is useful because the format
+	 * of a post varies between the retrieve and update endpoints. When
+	 * retrieving a post, the thumbnail ID is assigned in `post_thumbnail`, but
+	 * in creating a post, the thumbnail ID is assigned to `featured_image`.
+	 *
+	 * @param  {Object} post Post object
+	 * @return {Number}      The featured image ID
+	 */
+	getFeaturedImageId: function( post ) {
+		if ( ! post ) {
+			return;
+		}
+
+		if ( 'featured_image' in post && ! /^https?:\/\//.test( post.featured_image ) ) {
+			// Return the `featured_image` property if it does not appear to be
+			// formatted as a URL
+			return post.featured_image;
+		}
+
+		if ( post.post_thumbnail ) {
+			// After the initial load from the REST API, pull the numeric ID
+			// from the thumbnail object if one exists
+			return post.post_thumbnail.ID;
```

Comment:
Rather than guarding with the conditional, we could use `_.get` here.
This would also give us an opportunity to set a default return value, something like so:
`return get( post, 'post_thumbnail.ID', null );`